src/OpenCL.cpp

```diff
@@ -394,30 +394,30 @@ void OpenCL_Network::forward(const std::vector<float>& input,
     constexpr size_t one_plane = width * height * sizeof(float);
 
     opencl.ensure_thread_initialized();
-    const size_t midSize = one_plane * Network::MAX_CHANNELS;
     const size_t inSize = sizeof(float) * input.size();
+    const size_t channels = input.size() / (width * height);
     const size_t finalSize = m_layers.back().outputs * one_plane;
 
+    // Input layers in groups of 2 or channels in groups of 8.
+    const size_t maxOutputs = std::max(size_t(Network::INPUT_CHANNELS >> 1),
+                                       channels >> 3);
+
     if (!opencl_thread_data.m_buffers_allocated) {
-        size_t alloc_midSize = one_plane * Network::MAX_CHANNELS;
```

Comment:
FWIW the reason this was historically hardcoded was that the first layer could have less outputs than the subsequent layers, so sizing based on the first call didn't work. But for the Zero design it works fine.